src/sentry/event_manager.py

```diff
@@ -2437,3 +2443,11 @@ def save_transaction_events(jobs: Sequence[Job], projects: ProjectsMapping) -> S
     _eventstream_insert_many(jobs)
     _track_outcome_accepted_many(jobs)
     return jobs
+
+
+@metrics.wraps("event_manager.save_generic_events")
+def save_generic_events(jobs: Sequence[Job], projects: ProjectsMapping) -> Sequence[Job]:
+    for job in jobs:
+        job["event"] = _get_event_instance(job["data"], project_id=job["project_id"])
```

Comment:
We need to save to nodestore here and a few other things, or were you planning on doing that in a separate pr?

Reply:
I can do that here. I'll reach out to see what else I need to add back